src/Corax/IndexWriter.cs

```diff
@@ -125,27 +146,82 @@ public long GetNumberOfEntries()
             return Transaction.LowLevelTransaction.RootObjects.ReadInt64(IndexWriter.NumberOfEntriesSlice) ?? 0;
         }
 
-        private void InsertToken(ByteStringContext context, ref IndexEntryReader entryReader, int tokenField, Dictionary<Slice, List<long>> field, long entryId)
+        [SkipLocalsInit]
+        private unsafe void InsertAnalyzedToken(ByteStringContext context, ref IndexEntryReader entryReader, int tokenField, Dictionary<Slice, List<long>> field, long entryId)
         {
+            _analyzer.GetOutputBuffersSize(512, out int bufferSize, out int tokenSize);
+
+            byte* tempWordsSpace = stackalloc byte[bufferSize];
+            Token* tempTokenSpace = stackalloc Token[tokenSize];
```

Comment:
Do we have to worry about *very* large allocations here?

Reply:
We may need to, but I am not worried about that yet. See comment on the 512  bytes.

Reply:
Hm... double checked, looked like the limit is here:

https://github.com/ravendb/ravendb/blob/feature/corax/src/Voron/Data/CompactTrees/CompactTree.cs#L646

`1024` - but that is probably the hard limit from the storage, we are fine with `512` bytes as the max term length